src/main/java/org/takes/rq/RqHref.java

```diff
@@ -72,19 +76,22 @@ public Href href() throws IOException {
                 .header("host").iterator();
             final Iterator<String> protos = new RqHeaders.Base(this)
                 .header("x-forwarded-proto").iterator();
-            final String host;
+            Text host = new TextOf("localhost");
             if (hosts.hasNext()) {
-                host = hosts.next().trim();
-            } else {
-                host = "localhost";
+                host = new Trimmed(new TextOf(hosts.next()));
             }
-            final String proto;
+            Text proto = new TextOf("http");
             if (protos.hasNext()) {
-                proto = protos.next().trim();
-            } else {
-                proto = "http";
+                proto = new Trimmed(new TextOf(protos.next()));
             }
-            return new Href(String.format("%s://%s%s", proto, host, uri));
+            return new Href(
+                String.format(
```

Comment:
@fanifieiev Use `FormattedText` here

Reply:
@paulodamaso That is under another issue: https://github.com/yegor256/takes/issues/1009